src/garage/envs/bullet/bullet_env.py

```diff
@@ -128,18 +222,19 @@ def __getstate__(self):
             dict: The instance’s __init__() arguments
 
         """
-        env = self._env.env
+        env = self.env.env
 
         # Extract constructor signature
         sig = inspect.signature(env.__init__)
         args = {}
         param_names = list(sig.parameters.keys())
 
         # Hard fix for args/private variable name inconsistency
-        if isinstance(env, (MinitaurBulletEnv, MinitaurBulletDuckEnv)):
+        if env.spec.id.find('MinitaurBulletEnv') >= 0 or env.spec.id.find(
+                'MinitaurBulletDuckEnv') >= 0:
             args['render'] = env._is_render
             param_names.remove('render')
-        elif issubclass(type(env), MJCFBaseBulletEnv):
+        elif any(env.spec.id.find(id) >= 0 for id in MJCFBASED_BULLET_ENV):
```

Comment:
why not `elif env.spec.id in MJCFBASED_BULLET_ENV:`?

Reply:
`spec.id` has the `-v0` suffix. Similar for the check of `RacecarZedBulletEnv`. An alternative is to hardcode the complete env-id in `MJCFBASED_BULLET_ENV` (e.v. `RacecarZedBulletEnv-v0`). But if py_bullet changes the env id this might not work.

Reply:
```python
>>> f = 'foo'
>>> g = 'foo-v1'
>>> f in g
True
>>>
```